model/src/main/java/com/cloudera/nav/sdk/model/entities/Entity.java

```diff
@@ -344,4 +351,10 @@ public boolean equals(Object o) {
   public int hashCode() {
     return identity.hashCode();
   }
+
+  public Boolean getIsEndPoint() { return isEndPoint; }
+
+  public void setIsEndPoint(Boolean isEndPoint) {
```

Comment:
ok, I thought about why this pattern felt weird. What you're doing is adding a transient attribute to control a completely different level of abstraction that really has nothing to do with the Entity itself. Now the Entity not only has to know about itself, it must also contain knowledge about how the SDK client will convert data during transport.